chain/validation/lotus_test.go

```diff
@@ -52,5 +58,71 @@ func TestLotusExample(t *testing.T) {
 	drv.AssertBalance(bob, 50)
 	// This should become non-zero after gas tracking and payments are integrated.
 	drv.AssertBalance(miner, uint64(expectedGasUsed))
+}
+
+func TestLotusMinerCreate(t *testing.T) {
+	factory := NewFactories()
+	drv := suites.NewStateDriver(t, factory.NewState())
+	gasPrice := big.NewInt(1)
+	gasLimit := state.GasUnit(2000) // needs to be over ~1100 for lotus operations
+	TotalNetworkBalance := big.NewInt(1).Mul(big.NewInt(TotalFilecoin), big.NewInt(0).SetUint64(FilecoinPrecision))
+
+	// create the init actor
+	initActorAddress, err := state.NewIDAddress(0)
+	require.NoError(t, err)
+	_, _, err = drv.State().SetActor(initActorAddress, state.InitActorCodeCid, big.NewInt(0))
+	require.NoError(t, err)
 
+	// then create network address
+	networkAddress, err := state.NewIDAddress(1)
+	require.NoError(t, err)
+	_, _, err = drv.State().SetActor(networkAddress, state.AccountActorCodeCid, TotalNetworkBalance)
+	require.NoError(t, err)
+
+	// lastly, create storage market actor
+	storageMarketActorAddress, err := state.NewIDAddress(2)
+	require.NoError(t, err)
+	_, _, err = drv.State().SetActor(storageMarketActorAddress, state.StorageMarketActorCodeCid, big.NewInt(0))
+	require.NoError(t, err)
+
+
+	// miner that mines in this test
+	testMiner := drv.NewAccountActor(0)
+	// account that will own the miner
+	minerOwner := drv.NewAccountActor(20000000000)
+
+	producer := chain.NewMessageProducer(factory.NewMessageFactory(drv.State()), gasLimit, gasPrice)
+
+	sectorSize := big.NewInt(16 << 20)
+	publicKey := []byte{1} // lotus does not follow spec wrt miner publicKey
+
+	peerID := RequireIntPeerID(t, 1)
+	bpid, err := peerID.MarshalBinary()
+	require.NoError(t, err)
+
+	msg, err := producer.StoragePowerCreateStorageMiner(minerOwner, 0, minerOwner, publicKey, sectorSize, bpid, chain.Value(1000000))
```

Comment:
This isn't the right call because Lotus doesn't have a storage power actor. For impl-specific message sugar, we have to put functions in the implementation that call to `producer.Build()`. They should use method ids that are above `chain.MethodCount` (that are then mapped to the right internal method id by the factory).